test/e2e/utils/bundles.go

```diff
@@ -0,0 +1,164 @@
+	if err := cmd.Run(); err != nil {
+		return err
+	}
+	return nil
+}
+
+func getOCIImage(dir, destImage string) (types.ImageReference, error) {
+	imageRef, err := ocilayout.ParseReference(fmt.Sprintf("%s:%s", dir, destImage))
+	if err != nil {
+		return imageRef, fmt.Errorf("could not create OCI image reference: %v", err)
+	}
+	return imageRef, nil
+}
+
+// Builds the bundle image onto local filesystem
+func createLocalBundleImage(imageName, targetImageDir string, bundleContentDirectories []string, labels map[string]string) (string, error) {
+	if len(imageName) == 0 {
+		return "", fmt.Errorf("could not create OCI image: empty image name")
+	}
+
+	ts := time.Now().Unix()
+	if len(targetImageDir) == 0 {
+		targetImageDir = defaultImageDir
+	}
+	containerName := fmt.Sprintf("%s-builder-%d", imageName, ts)
+
+	if cwd, err := os.Getwd(); !path.IsAbs(targetImageDir) && err == nil {
+		targetImageDir = path.Join(cwd, targetImageDir)
+	}
+	layerCacheDir := path.Join(targetImageDir, ".cache")
+
+	if _, err := os.Stat(targetImageDir); os.IsNotExist(err) {
+		_ = os.MkdirAll(targetImageDir, 0755)
+	}
+	if _, err := os.Stat(layerCacheDir); os.IsNotExist(err) {
+		_ = os.MkdirAll(layerCacheDir, 0755)
+	}
+
+	imageRef, err := getOCIImage(targetImageDir, imageName)
+	if err != nil {
+		return "", fmt.Errorf("error parsing image reference: %v", err)
+	}
+
+	if buildah.InitReexec() {
+		return "", fmt.Errorf("buildah exec already initialized")
+	}
+	unshare.MaybeReexecUsingUserNamespace(false)
+
+	storeOptions, err := storage.DefaultStoreOptions(unshare.IsRootless(), unshare.GetRootlessUID())
+	if err != nil {
+		return "", fmt.Errorf("could not get default store options: %v", err)
+	}
+
+	localStore, err := storage.GetStore(storeOptions)
+	if err != nil {
+		return "", fmt.Errorf("could not get default store: %v", err)
+	}
+
+	b, err := buildah.NewBuilder(context.TODO(), localStore, buildah.BuilderOptions{
+		FromImage:        "scratch",
+		Container:        containerName,
+		ConfigureNetwork: buildah.NetworkDefault,
+		ReportWriter:     os.Stderr,
+		SystemContext:    &types.SystemContext{},
+		CommonBuildOpts:  &buildah.CommonBuildOptions{},
+	})
+	if err != nil {
+		return "", fmt.Errorf("error creating image builder: %v", err)
+	}
+	defer func() {
+		_ = b.Delete()
+	}()
+
+	for k, v := range labels {
+		b.SetLabel(k,v)
+	}
+
+	for _,copydir := range bundleContentDirectories {
+		if err := b.Add(copydir, false, buildah.AddAndCopyOptions{}, copydir); err != nil {
+			return "", fmt.Errorf("error adding layer: %v", err)
+		}
+	}
+
+	var dest types.ImageReference
+	dest, err = blobcache.NewBlobCache(imageRef, layerCacheDir, types.PreserveOriginal)
+	_, _, _, err = b.Commit(context.TODO(), dest, buildah.CommitOptions{
+		ReportWriter:          os.Stderr,
+		SystemContext: &types.SystemContext{
+		},
+		OmitTimestamp:         true,
+	})
+	if err != nil {
+		return "", fmt.Errorf("error creating image from container %s: %v", containerName, err)
+	}
+
```

Comment:
would potentially be cool to validate the image via `opm validate ...` somewhere here as well

Reply:
Added validation support